src/utils/testing/TimeMachine.ts

```diff
@@ -25,6 +25,7 @@ export class TimeMachine {
      */
     public forward(ms: number): void {
         jest.advanceTimersByTime(ms);
+        this.date += ms;
```

Comment:
General question: When I wrote this class, I remember testing this function to make sure `jest.advanceTimersByTime()` actually advanced the date as well. I'm pretty sure it did. Does it not actually?

Reply:
Just want to avoid a situation where we're unintentionally advancing the date by twice the value.

Reply:
It wasn't advancing it in such a way that would cause `AppTimer` (at the time) to fire its events, at least not now that it's been refactored. This one change caused those events to fire properly and tests to pass, without any visible consequences in terms of test failures. It would make sense for everything to work without manually incrementing the date, since `get date()` just pulls current millis anyway, so I'm not sure why those tests failed with no changes other than refactoring.